marlowe-runtime/test/Language/Marlowe/Runtime/Transaction/BuildConstraintsSpec.hs

```diff
@@ -223,3 +236,88 @@ withdrawSpec = Hspec.describe "buildWithdrawConstraints" do
+    let
+      -- System start and era history are reusable across tests.
+      systemStart = SystemStart $ posixSecondsToUTCTime 0  -- Without loss of generality.
+      eraParams = EraParams
+        { eraEpochSize = 1
+        , eraSlotLength = mkSlotLength 1
+        , eraSafeZone = UnsafeIndefiniteSafeZone
+        }
+      oneSecondBound i = Bound
+        { boundTime = RelativeTime $ fromInteger i
+        , boundSlot = fromInteger i
+        , boundEpoch = fromInteger i
+        }
+      oneSecondEraSummary i = EraSummary
+        { eraStart = oneSecondBound i
+        , eraEnd = EraEnd $ oneSecondBound $ i + 1
+        , eraParams
+        }
+      unboundedEraSummary i = EraSummary
+        { eraStart = oneSecondBound i
+        , eraEnd = EraUnbounded
+        , eraParams
+        }
+      eraHistory =
+        EraHistory CardanoMode
+          . mkInterpreter
+          . summaryWithExactly
+          $ Exactly
+          $  K (oneSecondEraSummary 0) -- Byron lasted 1 second
+          :* K (oneSecondEraSummary 1) -- Shelley lasted 1 second
+          :* K (oneSecondEraSummary 2) -- Allegra lasted 1 second
+          :* K (oneSecondEraSummary 3) -- Mary lasted 1 second
+          :* K (oneSecondEraSummary 4) -- Alonzo lasted 1 second
+          :* K (unboundedEraSummary 5) -- Babbage never ends
+          :* Nil
+    Hspec.QuickCheck.prop "valid slot interval for timed-out contract" do
+      address <- arbitrary
+      assets <- arbitrary
+      utxo <- arbitrary
+      -- The choice intervals for the tip, minimum time, and timeout overlap, so every ordering will occur.
+      tipSlot' <- chooseInteger (9, 20) -- Make sure the tip is in the Babbage era.
+      minTime <- oneof
+                   [
+                     (1000 *) <$> chooseInteger (6, 20)  -- Even seconds, so there is a chance of collision.
+                   , chooseInteger (6000, 20000)         -- Milliseconds, so there is rounding off.
+                   ]
+      timeout <- oneof
+                   [
+                     (1000 *) <$> chooseInteger (6, 20)  -- Even seconds, so there is a chance of collision.
+                   , chooseInteger (6000, 20000)         -- Milliseconds, so there is rounding off.
+                   ]
+      marloweParams <- arbitrary
+      let
+        toSlot = (`div` 1000)
+        tipSlot = toEnum . fromEnum $ tipSlot'
+        tipTime = 1000 * tipSlot'
+        marloweState = Semantics.State AM.empty AM.empty AM.empty $ POSIXTime minTime
+        marloweContract = Semantics.When [] (POSIXTime timeout) Semantics.Close  -- This contract can only time out.
+        datum = Semantics.MarloweData{..}
+        marloweOutput = TransactionScriptOutput{..}
+        result =
+          buildApplyInputsConstraints
+            systemStart eraHistory MarloweV1
+            marloweOutput
+            tipSlot
+            (Chain.TransactionMetadata mempty) Nothing Nothing mempty
+      pure
+        case result of
```

Comment:
Not necessary but Pablo had shown me how to use `label` and `counterexample` in these checks so it produces little tables of the distribution of cases. Handy to see how well the generators are covering or if they're producing cases at all! You can see use of this in tests I wrote in `marlowe-runtime/test/Language/Marlowe/Runtime/Transaction/ConstraintsSpec.hs`

Reply:
They look like this

``` 
sufficient collateral is selected if possible [v]
  +++ OK, passed 100 tests:
  85% Wallet has funds, selection succeeded
  15% Wallet does not have funds, selection failed
```

Reply:
Thanks, I added the `counterexample`s.